eng/common/scripts/TypeSpec-Project-Process.ps1

```diff
@@ -0,0 +1,115 @@
+# For details see https://github.com/Azure/azure-sdk-tools/blob/main/doc/common/TypeSpec-Project-Scripts.md
+
+[CmdletBinding()]
+param (
+  [Parameter(Position = 0)]
+  [ValidateNotNullOrEmpty()]
+  [string] $TypeSpecProjectDirectory, # A directory of `tspconfig.yaml` or a remoteUrl of `tspconfig.yaml`
+  [Parameter(Position = 1)]
+  [string] $CommitHash,
+  [Parameter(Position = 2)]
+  [string] $RepoUrl
+)
+
+. $PSScriptRoot/common.ps1
+if ($RepoUrl -and !$RepoUrl.StartsWith("https://github.com") -and ($RepoUrl.Split("/").Length -lt 4)) {
+  Write-Host "Parameter 'RepoUrl' has incorrect value: $RepoUrl. It should be similar like 'https://github.com/Azure/azure-rest-api-specs'"
+  exit
+}
+
+function CreateUpdate-TspLocation() {
+  $serviceDir = ""
+  $additionalDirs = @()
+
+  # Parse tspcofig.yaml to get service-dir, additionalDirectories, and package-dir
+  if ($tspConfigYaml["parameters"] -and $tspConfigYaml["parameters"]["service-dir"]) {
+    $serviceDir = $tspConfigYaml["parameters"]["service-dir"]["default"];
+  }
+  else {
+    Write-Host "Missing service-dir in parameters section of tspconfig.yaml."
```

Comment:
Write-Error and we should exit 1 so it is a non-zero exit code so callers will know this failed.